File: CodeFestival/Q31-Q40/Q35_Factory_함수_이용하기.js
function one(n){
    function two(m){
        let mul = m;
        for(let i=0; i<n-1; i++) {
            m = m * mul;
        }
        return m;
    }
    return two;
}

const a = one(2);
const b = one(3);
const c = one(4);

console.log(a(10));
console.log(b(10));
console.log(c(10));

// Math.pow 사용하면 제곱가능.

// function one(n) {
//     function two(value) {
//       const sq = Math.pow(value, n);
//       return sq;
//     }
//     return two;
//   }
  
//   const a = one(2);
//   const b = one(3);
//   const c = one(4);
  
//   console.log(a(10));
//   console.log(b(10));
//   console.log(c(10));

File: CodeFestival/Q31-Q40/Q35_Factory_함수_이용하기.py
def one(n):
    def two(m):
        mul = m
        for i in range(0, n-1, 1):
            m = m * mul
        return m    
    return two

a = one(2)
b = one(3)
c = one(4)
print(a(10))
print(b(10))
print(c(10))

# ** 사용하면 제곱 가능

# def one(n):
#     def two(value):
#         sq = value ** n
#         return sq
#     return two

# a = one(2)
# b = one(3)
# c = one(4)
# print(a(10))
# print(b(10))
# print(c(10))